src/iter/test.rs

```diff
@@ -1608,3 +1608,14 @@ fn check_partition_map() {
     assert_eq!(a, vec![1, 2, 3]);
     assert_eq!(b, "abcxyz");
 }
+
+#[test]
+fn check_repeat_eq() {
+    let v = vec!(4,4,4,4);
+    let mut fours: Vec<_> = repeat(4)
+    .take(4)
+    .zip(v)
+    .collect();
```

Comment:
Is there something I should do in order to get take to work effectively with this implementation. It seems to play nicely with others, but no so nicely by itself.